core/src/main/java/org/apache/iceberg/util/SortOrderUtil.java

```diff
@@ -37,16 +37,14 @@ private SortOrderUtil() {
   }
 
   public static SortOrder buildSortOrder(Table table) {
-    return buildSortOrder(table.spec(), table.sortOrder());
+    return buildSortOrder(table.schema(), table.spec(), table.sortOrder());
   }
 
-  public static SortOrder buildSortOrder(PartitionSpec spec, SortOrder sortOrder) {
```

Comment:
I may need to keep the old signature for compatibility but deprecate it. It is a util class, though.